diff -u a/lib/rubygems/path_support.rb b/lib/rubygems/path_support.rb
--- a/lib/rubygems/path_support.rb	2012-09-10 12:49:16.607647961 +0200
+++ b/lib/rubygems/path_support.rb	2012-10-01 14:39:35.931810732 +0200
@@ -52,6 +52,17 @@
         gem_path.push(*gpaths)
       else
         gem_path.push(*gpaths.split(File::PATH_SEPARATOR))
+
+        0.upto(gem_path.size - 1) do |i|
+          # Rejoin incorrectly split URLs. Push proto + ':' onto following item
+          if gem_path[i] =~ /^(jar(:file)?|file|classpath)$/ || # recognize commonly encountered protocols
+              (gem_path[i] =~ /^[a-z]+$/ && # some other protocol and following element begins with '//'
+               gem_path[i+1] && gem_path[i+1] =~ %r{^//})
+            gem_path[i+1] = gem_path[i] + ':' + gem_path[i+1]
+            gem_path[i] = nil
+          end
+        end if File::PATH_SEPARATOR == ':'
+        gem_path.compact!
       end
 
       if File::ALT_SEPARATOR then
